test/__tests__/data/klaviyo.json

```diff
@@ -658,6 +818,129 @@
       "files": {}
     }
   },
+  {
+    "description": "Track event call with flatten properties enabled",
+    "input": {
+      "destination": {
+        "Config": {
+          "publicApiKey": "WJqij9",
+          "privateApiKey": "pk_b68c7b5163d98807fcb57e6f921216629d",
+          "flattenProperties": true
+        }
+      },
+      "message": {
+        "type": "track",
+        "event": "TestEven001",
+        "sentAt": "2021-01-25T16:12:02.048Z",
+        "userId": "sajal12",
+        "channel": "mobile",
+        "context": {
+          "os": {
+            "name": "Android",
+            "version": "10"
+          },
+          "app": {
+            "name": "KlaviyoTest",
+            "build": "1",
+            "version": "1.0",
+            "namespace": "com.rudderstack.android.rudderstack.sampleAndroidApp"
+          },
+          "device": {
+            "id": "9c6bd77ea9da3e68",
+            "name": "raphaelin",
+            "type": "android",
+            "model": "Redmi K20 Pro",
+            "manufacturer": "Xiaomi"
+          },
+          "locale": "en-IN",
+          "screen": {
+            "width": 1080,
+            "height": 2210,
+            "density": 440
+          },
+          "traits": {
```

Comment:
we do have populateCustomFieldsFromTraits - add few traits which are not in `WhiteListedTraits` for the test case

Reply:
In general, the same traits under identify will be part of track event - so again populating them in the track properties too is expected behaviour from Klaviyo ?
cc: @utsabc

Reply:
To take as backlog item